plugins/database/src/db_plugin.cpp

```diff
@@ -5225,7 +5263,45 @@ irods::error db_mod_zone_op(
         }
         OK = 1;
     }
-    if ( strcmp( _option, "conn" ) == 0 ) {
+    if (strcmp(_option, "conn") == 0) {
+        // =-=-=-=-=-=-=-
+        // validate the connection string is well formed
+        const std::size_t addr_len = std::strlen(_option_value);
+        if (addr_len > 0) {
+            rodsHostAddr_t addr{};
+            // _option_value is const, so copy it
+            auto addr_buf = std::make_unique<char[]>(addr_len + 1);
+            std::strncpy(addr_buf.get(), _option_value, addr_len + 1);
+            status = parseHostAddrStr(addr_buf.get(), &addr);
+            if (status < 0) {
+                log_db::info("chlModZone failed to validate connection info [{}]", status);
+                addRErrorMsg(&_ctx.comm()->rError, status, "chlRegZone failed to validate connection info");
+                return ERROR(status, "failed to validate connection info");
+            }
+            const std::size_t host_len = std::strlen(addr.hostAddr);
+            if (host_len == 0 || (host_len == 1 && addr.hostAddr[0] == ':')) {
+                log_db::info("chlModZone failed to validate connection info due to empty hostname [{}]", _option_value);
+                addRErrorMsg(&_ctx.comm()->rError,
+                             CAT_HOSTNAME_INVALID,
+                             "chlRegZone failed to validate connection info due to empty hostname");
+                return ERROR(CAT_HOSTNAME_INVALID, "failed to validate connection info due to empty hostname");
+            }
+            if ((addr.portNum > 65535) || (addr.portNum <= 0)) {
+                log_db::info("chlModZone failed to validate connection info due to invalid or unspecified port [{}]",
+                             _option_value);
+                addRErrorMsg(&_ctx.comm()->rError,
+                             CAT_INVALID_ARGUMENT,
+                             "chlRegZone failed to validate connection info due to invalid or unspecified port");
+                return ERROR(CAT_INVALID_ARGUMENT,
+                             "failed to validate connection info due to invalid or unspecified port");
+            }
+        }
+        else {
+            log_db::info("connection info empty or not specified");
+            addRErrorMsg(&_ctx.comm()->rError, CAT_INVALID_ARGUMENT, "connection info empty or not specified");
+            return ERROR(CAT_INVALID_ARGUMENT, "connection info empty or not specified");
+        }
```

Comment:
Is this block identical to the previous one?

If so, consider moving it (or parts of it) into a function so the logic is shareable across the two locations.